### PR TITLE
Create CI strategy

### PR DESCRIPTION
**Pull request by martinkearn**
This PR adds the high-level strategy for CI pipelines across the whole Burokratt solution.

### CODE REVIEW

**Review by yahya130 — APPROVED** [review]

File: docs/technical/continuous-integration-strategy.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+CI pipelines should run as part of the PR process when merging features branches to `main`. A PR should have a "green" CI pipeline before code can be merged and the PR can be completed, this means that all automated checks in the CI pipeline must pass.
+
+## What does CI do?
+
+All CI pipelines will do the following tasks
+
+| Task             | Description                                                  |
+| ---------------- | ------------------------------------------------------------ |
+| dotnet format    | Uses the `dotnet` CLI to format the code according to defined style rules for the project. See https://github.com/dotnet/format |
+| dotnet restore   | Uses the `dotnet` CLI to restore the .net packages which the project depends on. See https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-restore |
```
Comment (decodingahmed) [inline docs/technical/continuous-integration-strategy.md:22]:
```suggestion
| dotnet restore   | Uses the `dotnet` CLI to restore the .NET packages which the project depends on. See https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-restore |
```

File: docs/technical/continuous-integration-strategy.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+| dotnet build     | Uses the `dotnet` CII to build the project. See https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build |
```
Comment (decodingahmed) [inline docs/technical/continuous-integration-strategy.md:23]:
```suggestion
| dotnet build     | Uses the `dotnet` CLI to build the project. See https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build |
```

File: docs/technical/continuous-integration-strategy.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+| dotnet test      | Uses the `dotnet` CLI to run all tests associated with the project. This include integration and unit tests. See https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test |
```
Comment (decodingahmed) [inline docs/technical/continuous-integration-strategy.md:24]:
```suggestion
| dotnet test      | Uses the `dotnet` CLI to run all tests associated with the project. This includes integration and unit tests. See https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test |
```

File: docs/technical/continuous-integration-strategy.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+| Mutation testing | Runs the mutation testing package as defined in [unit-testing-strategy](unit-testing-strategy.md). |
+| dotnet publish   | Uses the `dotnet` CI to publish the project. This makes the files available for packaging. |
```
Comment (decodingahmed) [inline docs/technical/continuous-integration-strategy.md:26]:
```suggestion
| dotnet publish   | Uses the `dotnet` CLI to publish the project. This produces deployable binaries ready for deployment. |
```

File: docs/technical/continuous-integration-strategy.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+| Build image      | Uses the `docker` CLI to build the image for the project     |
+| Sign image       | Signs the docker image for the project                       |
+| Upload image     | Uploads the Docker image to the GitHub container registry    |
```
Comment (decodingahmed) [inline docs/technical/continuous-integration-strategy.md:29]: Just using consistent wording for uploading docker images (i.e. - 'push')
```suggestion
| Push image     | Pushes the Docker image to the GitHub container registry    |
```

**Review by decodingahmed — COMMENTED** [review]
Just added some review notes to be address in the next PR.

**Pull request merged**